src/manageiq_client/api.py

```diff
@@ -617,26 +617,27 @@ def __call__(self, *args, **kwargs):
         return outcome
 
     def _get_entity_from_href(self, result):
+        """Returns entity in correct collection.
+
+        If the "href" value in result doesn't match the current collection,
+        try to find the collection that the "href" refers to.
+        """
         href_result = result['href']
-        href_match = None
-
-        # If the "href" value in result doesn't match the current collection,
-        # try to find the collection that the "href" refers to.
-        if self.collection._href not in href_result:
-            href_match = re.match(r"(http[^?]+)/([a-z_-]+)", href_result)
-
-        if href_match:
-            collection_name = href_match.group(2)
-            entry_point = href_match.group(1)
-            if collection_name and entry_point:
-                new_collection = Collection(
-                    self.collection.api,
-                    "{}/{}".format(entry_point, collection_name),
-                    collection_name
-                )
-                return Entity(new_collection, result, incomplete=True)
 
-        return Entity(self.collection, result, incomplete=True)
+        if self.collection._href.startswith(href_result):
+            return Entity(self.collection, result, incomplete=True)
+
+        href_match = re.match(r"(https?://.+/api[^?]*)/([a-z_-]+)", href_result)
```

Comment:
@simon3z @cben To be sure that the data obtained from regex are meaningful (not that it just matched something) I changed this to more complex regex and raise exception if it doesn't match. This way it was also possible to simplify the rest of the logic.

I added more tests to make sure this works as expected in all corner cases.